src/jit_no_ir.h

```diff
@@ -2096,13 +2100,18 @@ class EmitX64 {
         compiling = false;
     }
     void callr(RelAddr7 addr, Cond cond) {
-        c.mov(dword[REGS + offsetof(JitRegisters, pc)], regs.pc);
+        const u32 ret_pc = regs.pc;
+        c.mov(dword[REGS + offsetof(JitRegisters, pc)], ret_pc);
         ConditionPass(cond, [&] {
             EmitPushPC();
             regs.pc += addr.Relative32();
             c.mov(dword[REGS + offsetof(JitRegisters, pc)], regs.pc);
         });
-        compiling = false;
+        // For static jump we can continue compiling.
+        compiling = cond.GetName() == CondValue::True;
+        if (compiling) {
+            call_stack.push(ret_pc);
```

Comment:
Might be good to check that call_stack size isn't getting enormous just in case too

Reply:
This is the biggest ret chain I saw by eye while looking at generated blocks. But yeah can add a check in code
[ret_chain.txt](https://github.com/GPUCode/teakra/files/14569076/ret_chain.txt)